admin/app/javascript/solidus_admin/tom-select/plugins/stash_on_search.js

```diff
@@ -0,0 +1,70 @@
+// Substitute default TomSelect functionality for single selects, where typing in the select field while there is
+//  an option selected shows both selected option and the search term.
+// Instead, hide the selected option when typing and bring it back when leaving the field.
+class Stash {
+  constructor() {
+    this.content = {};
```

Comment:
Nit: You could [make use of a private field here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties).